rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -460,6 +500,461 @@ MasterGraph::timing()
+                                    fB = _mm256_cvtepi32_ps(_mm256_shuffle_epi8(pix0, mask_R));
+                                    fG = _mm256_cvtepi32_ps(_mm256_shuffle_epi8(pix0, mask_G));
+                                    fR = _mm256_cvtepi32_ps(_mm256_shuffle_epi8(pix0, mask_B));
+                                    fB = _mm256_fmadd_ps(fB, pmul0, padd0);
+                                    fG = _mm256_fmadd_ps(fG, pmul1, padd1);
+                                    fR = _mm256_fmadd_ps(fR, pmul2, padd2);
+                                    tempB = _mm256_cvtps_ph(fB, _MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC);
+                                    tempG = _mm256_cvtps_ph(fG, _MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC);
+                                    tempR = _mm256_cvtps_ph(fR, _MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC);
+                                    _mm_storeu_si128((__m128i *)B_buf_16, tempB);
+                                    _mm_storeu_si128((__m128i *)G_buf_16, tempG);
+                                    _mm_storeu_si128((__m128i *)R_buf_16, tempR);
+                                    B_buf_16 += 8;
+                                    G_buf_16 += 8;
+                                    R_buf_16 += 8;
+                                    in_buffer += 24;
+                                }
+                                for (; i < channel_size; i++, in_buffer += 3) {
+                                    *B_buf_16++ = (half) (in_buffer[0] * multiplier0) + offset0;
+                                    *G_buf_16++ = (half) (in_buffer[1] * multiplier1) + offset1;
+                                    *R_buf_16++ = (half) (in_buffer[2] * multiplier2) + offset2;
+                                }
+        #else
+                                for (unsigned channel_idx = 0; channel_idx < c; channel_idx++) {
+                                    for (unsigned i = 0; i < channel_size; i++)
+                                        output_tensor_16[dest_buf_offset + channel_idx * channel_size + i] =
+                                                offset[channel_idx] + multiplier[channel_idx] *
+                                                                    (reverse_channels ? (half) (in_buffer[(c * i + c - channel_idx - 1)])
+                                                                                        : (half) (in_buffer[(c * i + channel_idx)]));
+                                }
+        #endif
+                            }
+                        }
+                    }  // NCHW or NHWC
+                } // for loop batch
+
+                dest_buf_offset_start += single_output_image_size;
+            }
+        }
+    }
+    _convert_time.end();
+    return Status::OK;
+}
+
+MasterGraph::Status
+MasterGraph::copy_output(unsigned char *out_ptr, size_t out_size_in_bytes)
+{
+    if(no_more_processed_data())
+        return MasterGraph::Status::NO_MORE_DATA;
+
+    if(_output_tensor_list.size() != 1)
```

Comment:
Why are we restricying 1 output for copy_output?

Reply:
This API may not work when more than one outputs are set. Because each output tensor may have different dimension in the tensor pipeline.
This API is mainly used for the trainings where only 1 output is set